src/allotropy/parsers/agilent_gen5/agilent_gen5_structure.py

```diff
@@ -1,26 +1,574 @@
+        measurement_labels = []
+
+        if read_mode == ReadMode.ABSORBANCE:
+            measurement_labels = cls._get_absorbance_measurement_labels(
+                label_prefix, device_control_data
+            )
+
+        if read_mode == ReadMode.FLUORESCENCE:
+            excitations = device_control_data.get(EXCITATION_KEY, [])
+            emissions = device_control_data.get(EMISSION_KEY, [])
+            measurement_labels = [
+                f"{label_prefix}{excitation},{emission}"
+                for excitation, emission in zip(excitations, emissions)
+            ]
+
+        if read_mode == ReadMode.LUMINESCENCE:
+            emissions = device_control_data.get(EMISSION_KEY)
+            for emission in emissions:
+                label = "Lum" if emission == "Full light" else emission
+                measurement_labels.append(f"{label_prefix}{label}")
 
-        if not plates:
-            msg = "No plate data found in file."
+        return measurement_labels
+
+    @classmethod
+    def _get_absorbance_measurement_labels(
+        cls, label_prefix: Optional[str], device_control_data: dict
+    ) -> list:
+        wavelengths = device_control_data.get(WAVELENGTHS_KEY, [])
+        pathlength_correction = device_control_data.get(PATHLENGTH_CORRECTION_KEY)
+        measurement_labels = []
+
+        for wavelenght in wavelengths:
+            label = f"{label_prefix}{wavelenght}"
+            measurement_labels.append(label)
+
+        if pathlength_correction:
+            test, ref = pathlength_correction.split(" / ")
+            test_label = f"{label_prefix}{test} [Test]"
+            ref_label = f"{label_prefix}{ref} [Ref]"
+            measurement_labels.extend([test_label, ref_label])
+
+        return measurement_labels
+
+    @classmethod
+    def _get_device_control_data(
+        cls, procedure_details: str, read_mode: ReadMode
+    ) -> dict:
+        list_keys = frozenset(
+            {
+                EMISSION_KEY,
+                EXCITATION_KEY,
+                OPTICS_KEY,
+                GAIN_KEY,
+                MIRROR_KEY,
+                WAVELENGTHS_KEY,
+            }
+        )
+        read_data_dict: dict = {label: [] for label in list_keys}
+        read_lines: list[str] = procedure_details.splitlines()
+        datum_len = 2
+
+        for line in read_lines:
+            strp_line = str(line.strip())
+            if strp_line.startswith("Read\t"):
+                read_data_dict["Step Label"] = cls._get_step_label(line, read_mode)
+                continue
+
+            elif strp_line.startswith(WAVELENGTHS_KEY):
+                wavelengths = strp_line.split(":  ")
+                read_data_dict[WAVELENGTHS_KEY].extend(wavelengths[1].split(", "))
+                continue
+
+            line_data: list[str] = strp_line.split(",  ")
+            for read_datum in line_data:
+                splitted_datum = read_datum.split(": ")
+                if len(splitted_datum) != datum_len:
```

Comment:
```suggestion
                if len(splitted_datum) != 2:
```

There isn't a good reason for pulling this out into a "constant", it seems to make the code less clear to me. If lint was complaining about this, ignore the rule on this line and I'll investigate.

Reply:
yes, this was a complain by the linter... I agree with you it is not necessary to use a variable for this.